doc/decisions/example_migrations/one-table-per-taxonomy-view-functions/2023-03-09-194135_plant_relations/up.sql

```diff
@@ -0,0 +1,165 @@
+CREATE SEQUENCE plants_id_seq
+AS INTEGER
+START WITH 1
+INCREMENT BY 1
+NO MINVALUE
+NO MAXVALUE
+CACHE 1;
+
+-- Create the "family" table
+CREATE TABLE families (
+    id INTEGER PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    property1 TEXT,
+    CONSTRAINT family_name_key UNIQUE (name)
+);
+
+-- Create the "genus" table
+CREATE TABLE genera (
+    id INTEGER PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    property1 TEXT,
+    family_id INTEGER REFERENCES families (id),
+    CONSTRAINT genus_name_key UNIQUE (name)
+);
+
+-- Create the "species" table
+CREATE TABLE species (
+    id INTEGER PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    property1 TEXT,
+    genus_id INTEGER REFERENCES genera (id),
+    CONSTRAINT species_name_key UNIQUE (name)
+);
+
+-- Create the "variety" table
+CREATE TABLE varieties (
+    id INTEGER DEFAULT nextval('plants_id_seq') PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    unique_name TEXT NOT NULL,
+    common_name_en TEXT [],
+    common_name_de TEXT [],
+    property1 TEXT,
+    species_id INTEGER REFERENCES species (id),
+    CONSTRAINT variety_name_key UNIQUE (unique_name),
+    CONSTRAINT variety_name_key_rename UNIQUE (name)
+);
+
+-- Create the "cultivar" table
+CREATE TABLE cultivars (
+    id INTEGER DEFAULT nextval('plants_id_seq') PRIMARY KEY,
+    unique_name TEXT NOT NULL,
+    common_name_en TEXT [],
+    common_name_de TEXT [],
+    property1 TEXT,
+    species_id INTEGER REFERENCES species (id),
+    variety_id INTEGER REFERENCES varieties (id),
+    CONSTRAINT unique_name_key UNIQUE (unique_name),
+    CONSTRAINT check_variety_or_species CHECK (
+        variety_id IS NULL OR species_id IS NULL
+    )
+);
+
+-- Create a view joining the tables together
+-- COALESCE function accepts an unlimited number of arguments.
+-- It returns the first argument that is not null.
+
+--todo
```

Comment:
Please use english here, even if you just leave notes for yourself.
This would help non german speaking team members to gauge your progress more when reviewing draft PRs.